RecoEgamma/EgammaTools/interface/EgammaDNNHelper.h

```diff
@@ -65,6 +66,7 @@ namespace egammaTools {
     std::vector<uint> nInputs_;
 
     std::vector<std::unique_ptr<const tensorflow::GraphDef>> graphDefs_;
+    std::vector<std::unique_ptr<tensorflow::Session>> sessions_;
```

Comment:
Assuming `tensor flow::Session` is _const thread-safe_ this should be `std::vector<std::unique_ptr<const tensorflow::Session>>` as it is used across threads concurrently.

Reply:
IIRC `tensorflow::Session` interface is not `const`, even if it is thread safe and does not mutate the `Session` object (i.e. in principle the interface could be `const`) :(

Reply:
I'm about to push a commit moving this to the `tensorflow::SessionCache` (https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/TensorFlow/interface/TensorFlow.h#L189) which is returning `const Session*` .